### PR TITLE
fix: enter to submit

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
WEB-227

## Test Plan
- Check if modals can handle both button click and `Enter`  key to submit
  - Workspace
    - Create
    - Edit
  - Project
    - Create
    - Edit
  - Model
    - Create
   

## Commentary (optional)
 I didn't enable `Enter` to submit in modals for deletion since `delete` is a unchangeable action.

## Checklist
- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: webui/react/src/hooks/useModal/Project/useModalProjectMove.tsx [hunk 1]
```diff
@@ -124,28 +118,24 @@ const useModalProjectMove = ({ onClose, project }: Props): ModalHooks => {
...
-    [handleOk, modalContent],
-  );
+  }, [form, project.id, project.name, workspaces]);
+
+  const getModalProps = useMemo((): ModalFuncProps => {
+    return {
+      closable: true,
+      content: modalContent,
+      icon: null,
+      okButtonProps: { form: FORM_ID, htmlType: 'submit' },
```
Comment (EmilyBonar) [inline webui/react/src/hooks/useModal/Project/useModalProjectMove.tsx:128]: fix: not sure why, but moving projects doesn't work with the enter button

File: webui/react/src/hooks/useModal/Workspace/useModalWorkspaceEdit.tsx [hunk 2]
```diff
@@ -1,44 +1,51 @@
...
-          Name
-        </label>
-        <Input id="name" value={name} onChange={handleNameInput} />
-      </div>
+      <Form autoComplete="off" className={css.base} form={form} id={FORM_ID} layout="vertical">
+        <Form.Item
+          initialValue={workspace.name}
+          label="Workspace Name"
+          name="workspaceName"
+          rules={[{ message: 'Workspace name is required ', required: true }]}>
```
Comment (EmilyBonar) [inline webui/react/src/hooks/useModal/Workspace/useModalWorkspaceEdit.tsx:38]: nit: remove space at end of message

File: webui/react/src/hooks/useModal/Project/useModalProjectEdit.tsx [hunk 3]
```diff
@@ -1,55 +1,54 @@
...
-          </label>
-          <Input id="description" value={description} onChange={handleDescriptionInput} />
-        </div>
-      </div>
+      <Form autoComplete="off" className={css.base} form={form} id={FORM_ID} layout="vertical">
+        <Form.Item
+          initialValue={project.name}
+          label="Project Name"
+          name="projectName"
+          rules={[{ message: 'Name is required ', required: true }]}>
```
Comment (EmilyBonar) [inline webui/react/src/hooks/useModal/Project/useModalProjectEdit.tsx:37]: nit: remove space at end of message

**Review by EmilyBonar — CHANGES_REQUESTED** [review]
Main thing is to investigate the move project modal. You should also look at converting the move experiment modal. Finally, I think making it so the primary button isn't disabled is a (small) regression, so it would be good to add that back if you can.

File: webui/react/src/hooks/useModal/Project/useModalProjectMove.tsx (hunk 1 above)
Comment (keita-determined) [reply webui/react/src/hooks/useModal/Project/useModalProjectMove.tsx]: https://stackoverflow.com/questions/2968693/why-doesnt-hitting-enter-when-a-select-is-focused-submit-the-form
Enter to submit doesnt apply to `select`, so i reverted it.

**Review by EmilyBonar — APPROVED** [review]
Looks good. I don't think this is necessarily blocking, but I did notice this weird behavior where the height expands when an error message disappears.

[Uncategorized Experiments - Determined - 3 November 2022 - Watch Video](https://www.loom.com/share/75d6d743c42241aabaecf851b8a7413f)

**Comment by keita-determined** [comment]
> Looks good. I don't think this is necessarily blocking, but I did notice this weird behavior where the height expands when an error message disappears.
> 
> [Uncategorized Experiments - Determined - 3 November 2022 - Watch Video](https://www.loom.com/share/75d6d743c42241aabaecf851b8a7413f)

oh yes. it has been there before this PR. I'll make a ticket for it.
I assume editing antd-css causes this?

**Pull request merged**